contacts/tokens.py

```diff
@@ -0,0 +1,15 @@
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.utils import six
+
+
+class TokenGenerator(PasswordResetTokenGenerator):
+
+    def _make_hash_value(self, user, timestamp):
+        return (
+            six.text_type(user.pk) + six.text_type(timestamp) +
+            six.text_type(user.is_active)
+        )
+
+
+account_activation_token = TokenGenerator()
```

Comment:
What is this doing exactly?

Reply:
in this variable i generate a token and then used this token while sending email url see bold line below

message = render_to_string('contacts/acc_active_email.html', {
                'user': user,
                'domain': current_site.domain,
                'uid': urlsafe_base64_encode(force_bytes(user.pk)).decode(),
                **'token': account_activation_token.make_token(user),**
            })